docs/src/P3Scheme.md

```diff
@@ -0,0 +1,141 @@
+# P3 Scheme
+
+The `P3Scheme.jl` module implements the predicted particle properties
+ (P3) scheme for ice-phase microphysics developed by [MorrisonMilbrandt2015](@cite)
+
+The P3 scheme is a 2-moment, bulk scheme involving a
+ single ice-phase category with 4 degrees of freedom: total mass,
+ rime mass, rime volume, and number mixing ratios.
+
+Traditionally, cloud ice microphysics models use various predefined
+ categories to represent ice modes, but the P3 scheme sidesteps the
+ problem of prescribing transitions between ice categories by adopting
+ a single ice category. This simplification might
+ aid in attempts to constrain scheme parameters.
+
+## Assumed particle size distribution (PSD)
+
+Following [MorrisonMilbrandt2015](@cite), the scheme assumes a
+ gamma distribution for the concentration of particles per unit volume:
+
+```math
+N'(D) = N_{0} D^\mu \, e^{-\lambda \, D}
+```
+
+where:
+ - ``N'`` is the number concentration,
+ - ``D`` is the maximum particle dimension,
+ - ``N_0`` is the intercept parameter,
+ - ``\mu`` is the shape parameter,
+ - ``\lambda`` is the slope parameter.
+
+and ``\mu \ = 0.00191 \lambda \ ^{0.8} - 2`` for ``\mu \ \in (0,6)``.
+
+## Assumed particle mass relationships
+
+The mass (m) of particles as a function of maximum particle dimension (D)
+ is a piecewise function with variable breakpoints described
+ by the following table.
+
+| particle properties |      condition(s)     |    m(D) relation      |
+|:--------------------|:----------------------|:----------------------|
+|small, spherical ice | ``D < D_{th}`` | ``\frac{\pi}{6} \rho_i \ D^3`` |
+|large, unrimed ice   | ``q_{rim} = 0`` and ``D > D_{th}`` | ``\alpha_{va} \ D^{\beta_{va}}`` |
+|dense nonspherical ice | ``q_{rim} > 0`` and ``D_{th} < D < D_{gr}`` | ``\alpha_{va} \ D^{\beta_{va}}`` |
+|partially rimed ice | ``q_{rim} > 0`` and ``D > D_{cr}`` | ``\frac{\alpha_{va}}{1-F_r} D^{\beta_{va}}`` |
+|graupel (completely rimed, spherical)| ``q_{rim} > 0``and ``D_{gr} < D < D_{cr}`` | ``\frac{\pi}{6} \rho_g \ D^3`` |
+
+where:
+ - ``\rho_i \ = 917 kgm^{-3}`` (density of bulk ice)
```

Comment:
I would suggest using whatever ice density we have in CLIMAParameters.

Reply:
Also, thats a lot of parameters. Maybe it would be good to group them a little. There are some that are physical constants (like ice density), some that are some empirical coefficients, some that are model state variables and some that are the outputs of the NL solve.

It would also be good to mention here briefly that you need the NL solver to get some of those values.

Reply:
Will do!